File: infrastructures/ddd/csharp/src/Infrastructure/Persistence/Read/Infra/BaseReadEntity.cs
using System;

namespace OddlyDdd.Infrastructure.Persistence.Read.Infra
{
    /// <summary>
    /// Base class for all read entities (query side of CQRS).
    /// Read entities are pre-rendered views optimized for front-end consumption.
    /// They are denormalized for query performance and may aggregate multiple write entities.
    /// </summary>
    public abstract class BaseReadEntity
    {
        /// <summary>
        /// Unique identifier for the entity
        /// </summary>
        public string Id { get; set; } = string.Empty;

        /// <summary>
        /// Timestamp when the read entity was created
        /// </summary>
        public DateTime CreatedAt { get; set; }

        /// <summary>
        /// Timestamp when the read entity was last updated
        /// </summary>
        public DateTime UpdatedAt { get; set; }

        protected BaseReadEntity()
        {
            CreatedAt = DateTime.UtcNow;
            UpdatedAt = DateTime.UtcNow;
        }
    }
}

File: infrastructures/rest-ddd/java/src/main/java/com/example/restddd/infrastructure/persistence/read/BaseReadEntity.java
/**
 * Base Read Entity - For Queries (SELECT)
 * Pre-rendered views optimized for front-end.
 * Denormalized for query performance.
 */

package com.example.restddd.infrastructure.persistence.read;

public abstract class BaseReadEntity {
    /**
     * Abstract base class for all Read Entities (Query side).
     * 
     * Rules:
     * - MUST be in /infrastructure/persistence/read/
     * - MUST suffix with 'ReadEntity'
     * - Pre-rendered views optimized for queries
     * - Denormalized data (may aggregate from multiple WriteEntities)
     * - NO version tracking needed (read-only)
     */

    protected String m_id;

    public String getId() {
        return m_id;
    }

    public void setId(String p_id) {
        this.m_id = p_id;
    }
}
